src/backend/distributed/metadata/node_metadata.c

```diff
@@ -273,6 +273,19 @@ citus_add_node(PG_FUNCTION_ARGS)
 	 */
 	if (!nodeAlreadyExists)
 	{
+		WorkerNode *workerNode = FindWorkerNodeAnyCluster(nodeNameString, nodePort);
+
+		/*
+		 * If the worker is not marked as a coordinator, check that
+		 * the node is not trying to add itself
+		 */
+		if (workerNode->groupId != 0 && IsWorkerTheCurrentNode(workerNode))
+		{
+			ereport(ERROR, (errmsg("Node cannot add itself as a worker."),
+							errhint("Either try to add the node as a coordinator "
```

Comment:
should we give the full command? That is mostly very useful:
```
.... as coordinator SELECT citus_add_node(%s, %d, groupid:=0), nodeNameString,Nodeport)

Reply:
Maybe use citus_set_coordinator_host